.gitignore

```diff
@@ -133,461 +150,42 @@ external_src/oomph_arpack/debug.h
-demo_drivers/interaction/fsi_collapsible_channel/fsi_collapsible_channel_algebraic_adapt_TH
-demo_drivers/interaction/fsi_collapsible_channel/fsi_collapsible_channel_macro_CR
-demo_drivers/interaction/fsi_collapsible_channel/fsi_collapsible_channel_macro_TH
-demo_drivers/interaction/fsi_collapsible_channel/fsi_collapsible_channel_macro_adapt_CR
-demo_drivers/interaction/fsi_collapsible_channel/fsi_collapsible_channel_macro_adapt_TH
-demo_drivers/interaction/fsi_collapsible_channel/fsi_pseudo_solid_collapsible_channel_CR
-demo_drivers/interaction/fsi_collapsible_channel/fsi_pseudo_solid_collapsible_channel_TH
-demo_drivers/interaction/fsi_collapsible_channel/fsi_pseudo_solid_collapsible_channel_adapt_CR
-demo_drivers/interaction/fsi_collapsible_channel/fsi_pseudo_solid_collapsible_channel_adapt_TH
-demo_drivers/interaction/fsi_driven_cavity/fsi_driven_cavity_driver
-demo_drivers/interaction/fsi_driven_cavity/fsi_driven_cavity_segregated_driver
-demo_drivers/interaction/fsi_osc_ring/fsi_osc_ring
-demo_drivers/interaction/fsi_osc_ring/fsi_osc_ring_compare_jacs
-demo_drivers/interaction/fsi_osc_ring/osc_ring_alg
-demo_drivers/interaction/fsi_osc_ring/osc_ring_macro
-demo_drivers/interaction/linearised_fsi_pulsewave/linearised_fsi_pulsewave
-demo_drivers/interaction/linearised_poroelastic_fsi_pulsewave/linearised_poroelastic_fsi_pulsewave
-demo_drivers/interaction/pseudo_solid_collapsible_tube/pseudo_solid_collapsible_tube
-demo_drivers/interaction/pseudo_solid_collapsible_tube/pseudo_solid_collapsible_tube_work
-demo_drivers/interaction/pseudo_solid_fsi_channel_with_leaflet/fsi_channel_with_leaflet_precond
-demo_drivers/interaction/turek_flag/turek_flag
-demo_drivers/interaction/unsteady_vmtk_fsi/unsteady_vmtk_fsi
-demo_drivers/interaction/unstructured_adaptive_fsi/unstructured_adaptive_2d_fsi
-demo_drivers/interaction/unstructured_fsi/unstructured_two_d_fsi
-demo_drivers/interaction/unstructured_three_d_fsi/unstructured_three_d_fsi
-demo_drivers/interaction/vmtk_fsi/vmtk_fsi
-demo_drivers/linear_elasticity/periodic_load/periodic_load
-demo_drivers/linear_elasticity/periodic_load/refineable_periodic_load
-demo_drivers/linear_solvers/HypreSolver_test
-demo_drivers/linear_solvers/TrilinosSolver_test
-demo_drivers/linear_solvers/adv_diff_iterative_linear_solver_tester
-demo_drivers/linear_solvers/direct_solver_test
-demo_drivers/linear_solvers/driven_cavity
-demo_drivers/linear_solvers/two_d_linear_elasticity_with_simple_block_diagonal_preconditioner
-demo_drivers/linear_solvers/two_d_poisson_cg
-demo_drivers/linear_solvers/two_d_poisson_dense_lu
-demo_drivers/linear_wave/two_d_linear_wave/two_d_linear_wave
-demo_drivers/linear_wave/two_d_linear_wave/two_d_linear_wave_flux
-demo_drivers/linear_wave/two_d_linear_wave_adapt/two_d_linear_wave_adapt
-demo_drivers/linking/demo_code
+bin/minimal_mpi_test.out
+
+demo_drivers/heat_transfer_and_melting/OUTPUT
+demo_drivers/heat_transfer_and_melting/result_spring_contact_kuhn_tucker.dat
```

Comment:
Can we add a rule for *.dat and remove this (and all other dats?)

Reply:
Unfortunately not, there are a lot of raw (unpacked) `.dat` files in various folders (e.g. some `.dat` files just contain the number of iterations for convergence of a self-test using an iterative solver and do not require compressing). Ignoring them would break several self-tests. For this reason, I'm going to leaving this rule untouched.

Reply:
Ignoring them wouldn't remove already committed files or ignore changes, but I get what you're saying.

Reply:
Yep, but it would make future additions more difficult. It would force the user to use `git add --force` to get their files in; if the user is using a glob expression for the staging then this is likely to cause extra things, that shouldn't be added, to be added in.